presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -112,13 +112,21 @@ public CachingHiveMetastore(@ForCachingHiveMetastore HiveMetastore delegate, @Fo
                 config.getMetastoreCacheMaximumSize());
     }
 
-    public CachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Duration refreshInterval, long maximumSize)
+    public CachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Optional<Duration> refreshInterval, long maximumSize)
     {
         this(
                 delegate,
                 executor,
                 OptionalLong.of(cacheTtl.toMillis()),
-                refreshInterval.toMillis() >= cacheTtl.toMillis() ? OptionalLong.empty() : OptionalLong.of(refreshInterval.toMillis()),
+                refreshInterval
+                        .flatMap(interval -> {
+                            if (interval.toMillis() >= cacheTtl.toMillis()) {
```

Comment:
should we modify `io.prestosql.plugin.hive.metastore.cache.CachingHiveMetastore#newCacheBuilder` also. It contains similar check as here.

Reply:
Right, we can remove the condition here.